policy/modules/services/incus.te

```diff
@@ -0,0 +1,204 @@
+policy_module(incus)
+## <desc>
+##	<p>
+##	Allow incusd to mount filesystems inside
+##	lxc containers that might be required
+##	by some legacy init systems.
+##	</p>
+## </desc>
+gen_tunable(incus_lxc_legacy_mounts, false)
+
+########################################
+#
+# Declarations
+#
+
+# common attribute for all container domains
+# that may be used with incus
+attribute incus_container_domain;
+
+container_engine_domain_template(incusd)
+container_system_engine(incusd_t)
+type incusd_exec_t;
+
+container_engine_executable_file(incusd_exec_t)
+
+init_daemon_domain(incusd_t, incusd_exec_t)
+
+ifdef(`enable_mls',`
+	init_ranged_daemon_domain(incusd_t, incusd_exec_t, s0 - mls_systemhigh)
+')
+
+type incusc_t;
+type incusc_exec_t;
+
+container_engine_executable_file(incusc_exec_t)
+application_domain(incusc_t, incusc_exec_t)
+
+# other engines like docker have this, what will it do?
+#container_engine_domain_template(incusd_user)
+
+# user containers are not tested yet
+#container_user_engine(incusd_user_t)
+#application_domain(incusd_user_t, incusd_exec_t)
+
+#mls_trusted_object(incusd_user_t)
+
+type incusc_user_t;
+application_domain(incusc_user_t, incusc_exec_t)
+
+########################################
+#
+# incus daemon local policy
+
+allow incusd_t self:process getattr;
+allow incusd_t self:anon_inode { create map read write };
+allow incusd_t self:io_uring sqpoll;
+allow incusd_t self:netlink_kobject_uevent_socket create_socket_perms;
+allow incusd_t self:vsock_socket create_stream_socket_perms;
+dontaudit incusd_t self:capability sys_module;
+
+# incusd needs to manage the host network
+allow incusd_t self:rawip_socket { create getopt ioctl read setopt write };
+allow incusd_t self:tun_socket { create relabelfrom relabelto };
+
+# incusd wants to load netdev-phys*
+kernel_request_load_module(incusd_t)
+
+ifdef(`init_systemd',`
+	init_start_system(incusd_t)
+	init_stop_system(incusd_t)
+	init_get_system_status(incusd_t)
+	init_stop_generic_units(incusd_t)
+')
+
+# losetup required for instance creation
+dev_rw_loop_control(incusd_t)
+dev_rw_vhost(incusd_t)
+
+# comm="incusd" path="/run/systemd/resolve/stub-resolv.conf"
+container_read_tmpfs_files(incusd_t)
+
+fs_getattr_binfmt_misc_fs(incusd_t)
+fs_mount_binfmt_misc_fs(incusd_t)
+fs_getattr_binfmt_misc_dirs(incusd_t)
+fs_dontaudit_manage_fusefs_files(incusd_t)
+
+# here dontaudit would also work, but consuming more resources
+fs_watch_tmpfs_dirs(incusd_t)
+
+dnsmasq_domtrans(incusd_t)
+dnsmasq_signal(incusd_t)
+dnsmasq_kill(incusd_t)
+dnsmasq_signull(incusd_t)
```

Comment:
I think this should be optional policy; the Debian wiki seems to also imply that dnsmasq is only required for the *default* networking bridge.